### ISSUE TITLE
Segmentation fault happens when calling an R function.

### ISSUE THREAD

**Issue by bicycle1885** [JuliaInterop/RCall.jl#182]
Calling a function `ComBat` defined in the [sva](https://bioconductor.org/packages/release/bioc/html/sva.html) package results in a segmentation fault. This seems to be related to IO functions but I have no time to debug it now so let me file an issue. I'm using RCall.jl v0.7.1 on Julia v0.5.1. Calling this function in an R session doesn't cause this problem.

```
$ julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.1 (2017-03-05 13:25 UTC)
 _/ |\__'_|_|_|\__'_|  |
|__/                   |  x86_64-apple-darwin16.4.0

julia> using RCall

julia> R"""
       library(sva)
       ComBat(matrix(rnorm(6), nrow=3), batch=c(1, 2))
       """
WARNING: RCall.jl: Loading required package: mgcv
Loading required package: nlme
This is mgcv 1.8-16. For overview type 'help("mgcv-package")'.
Loading required package: genefilter
Found 2 batches
Note: one batch has only one sample, setting mean.only=TRUE
Adjusting for 0 covariate(s) or covariate level(s)
Standardizing Data across genes
Fitting L/S model and finding priors

signal (11): Segmentation fault: 11
while loading no file, in expression starting on line 72
uv_write2 at /Users/kenta/vendor/julia/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/unix/stream.c:1389
uv_write at /Users/kenta/vendor/julia/deps/srccache/libuv-8d5131b6c1595920dd30644cd1435b4f344b46c8/src/unix/stream.c:1475
jl_uv_write at /Users/kenta/vendor/julia/src/jl_uv.c:424
uv_write at ./stream.jl:809
unsafe_write at ./stream.jl:830
write at ./io.jl:175
#write_output#1 at /Users/kenta/.julia/v0.5/RCall/src/Console.jl:31
unknown function (ip: 0x1151441a0)
reval_p at /Users/kenta/.julia/v0.5/RCall/src/eval.jl:56
unknown function (ip: 0x115140d33)
jl_call_method_internal at /Users/kenta/vendor/julia/src/./julia_internal.h:210 [inlined]
jl_apply_generic at /Users/kenta/vendor/julia/src/gf.c:1950
reval_p at /Users/kenta/.julia/v0.5/RCall/src/eval.jl:79
reval_p at /Users/kenta/.julia/v0.5/RCall/src/eval.jl:90
unknown function (ip: 0x11515330c)
jl_call_method_internal at /Users/kenta/vendor/julia/src/./julia_internal.h:210 [inlined]
jl_apply_generic at /Users/kenta/vendor/julia/src/gf.c:1950
reval at /Users/kenta/.julia/v0.5/RCall/src/eval.jl:98
unknown function (ip: 0x115153286)
jl_call_method_internal at /Users/kenta/vendor/julia/src/./julia_internal.h:210 [inlined]
jl_apply_generic at /Users/kenta/vendor/julia/src/gf.c:1950
do_call at /Users/kenta/vendor/julia/src/interpreter.c:66
eval at /Users/kenta/vendor/julia/src/interpreter.c:190
eval_body at /Users/kenta/vendor/julia/src/interpreter.c:469
jl_interpret_call at /Users/kenta/vendor/julia/src/interpreter.c:573
jl_toplevel_eval_flex at /Users/kenta/vendor/julia/src/toplevel.c:572
jl_toplevel_eval_in_warn at /Users/kenta/vendor/julia/src/builtins.c:590
eval at ./boot.jl:234
jlcall_eval_19869 at /usr/local/julia/v0.5/lib/julia/sys.dylib (unknown line)
jl_call_method_internal at /Users/kenta/vendor/julia/src/./julia_internal.h:210 [inlined]
jl_apply_generic at /Users/kenta/vendor/julia/src/gf.c:1950
eval_user_input at ./REPL.jl:64
unknown function (ip: 0x11513c6d6)
jl_call_method_internal at /Users/kenta/vendor/julia/src/./julia_internal.h:210 [inlined]
jl_apply_generic at /Users/kenta/vendor/julia/src/gf.c:1950
macro expansion at ./REPL.jl:95 [inlined]
#3 at ./event.jl:68
unknown function (ip: 0x11513521f)
jl_call_method_internal at /Users/kenta/vendor/julia/src/./julia_internal.h:210 [inlined]
jl_apply_generic at /Users/kenta/vendor/julia/src/gf.c:1950
jl_apply at /Users/kenta/vendor/julia/src/./julia.h:1392 [inlined]
start_task at /Users/kenta/vendor/julia/src/task.c:254
Allocations: 4785375 (Pool: 4784308; Big: 1067); GC: 6
fish: 'julia' terminated by signal SIGSEGV (Address boundary error)
```

```
~/.j/v/RCall ((v0.7.1)|✔) $ R

R version 3.3.2 (2016-10-31) -- "Sincere Pumpkin Patch"
Copyright (C) 2016 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin14.5.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

>
> library(sva)
Loading required package: mgcv
Loading required package: nlme
This is mgcv 1.8-16. For overview type 'help("mgcv-package")'.
Loading required package: genefilter
> ComBat(matrix(rnorm(6), nrow=3), batch=c(1, 2))
Found 2 batches
Note: one batch has only one sample, setting mean.only=TRUE
Adjusting for 0 covariate(s) or covariate level(s)
Standardizing Data across genes
Fitting L/S model and finding priors
Finding parametric adjustments
Adjusting the Data
       1   2
[1,] NaN NaN
[2,] NaN NaN
[3,] NaN NaN
```

**Comment by randy3k** [comment]
it should be fixed once #183 is merged.

**Comment by bicycle1885** [comment]
Thanks a lot!
